playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java

```diff
@@ -212,8 +215,11 @@ boolean prepareTracks(TracksInfo trackSelections, String externalThumbnailWebVtt
         warnAboutUnsupportedRendererTypes();
 
         List<CustomFormat> rawImageTracks = new ArrayList<>();
+        List<EventStream> eventStreamsList = new ArrayList<>();
+
         if (customDashManifest != null) {
             for (int periodIndex = 0; periodIndex < customDashManifest.getPeriodCount(); periodIndex++) {
+                eventStreamsList.addAll(customDashManifest.getPeriod(periodIndex).eventStreams);
```

Comment:
need to verify are allowed /  can access the periodIndex without npe

Reply:
It is safe to call `getPeriod` because ExoPlayer internally creates an empty list. We are on the safe side.